CPack: .deb package

AYESDIE opened the pull request:
There are still changes that are to be made to this.

But in the meantime, to test things out, the following commands can be used.
```
mkdir build && cd build
cmake ../
make 
make build_images
make build_pofiles
make plugins_xml
make package
```
Once this is done, there will be a .deb file and a .tar.gz file in the build folder.
![image](https://user-images.githubusercontent.com/34449856/86362550-1367b700-bc93-11ea-955e-1e28536e4fe2.png)

Once .deb package is installed from either terminal or from UI, you can run Synfig Studio by running this on the command line.
```
synfigstudio
```


To-do:
- [x] Set up version properly.
- [x] Set up dependencies and the dependency version properly.
- [x] Set up an icon.
- [x] Set up a better description rather than *README.md*.
![image](https://user-images.githubusercontent.com/34449856/86362978-c0daca80-bc93-11ea-847f-937ab766ea73.png)

AYESDIE commented:
I added changes for #1402 here, so now it shows on menu. 
![image](https://user-images.githubusercontent.com/34449856/86448632-a3aa0880-bd34-11ea-8df4-cda6a36a7fb8.png)

P.S: Working on fixing that icon issue.

ice0 commented:
The problem with the icon is possible due to an incorrect icon name. May also be related to this https://github.com/synfig/synfig/issues/1412

Snap fixes:
https://github.com/synfig/synfig/blob/97940e960d81b7fd02c070bfd1d1416e5ab96b39/snap/snapcraft.yaml#L120

Flatpak:
https://github.com/flathub/org.synfig.SynfigStudio/blob/1409de8e34f107340aec0dfa15317a7bb3da3759/org.synfig.SynfigStudio.yaml#L220

Keyikedalube commented on synfig-studio/images/CMakeLists.txt:
```diff
@@ -272,6 +272,10 @@ endforeach()
 # Synfig Splash Screen
 if(${CMAKE_BUILD_TYPE} MATCHES Debug)
     set(SYNFIG_SPLASH_SCREEN ${CMAKE_CURRENT_BINARY_DIR}/splash_screen_development.sif)
+    file(
+        COPY ${CMAKE_CURRENT_SOURCE_DIR}/logo.sif
+        DESTINATION ${CMAKE_CURRENT_BINARY_DIR}
+    )
```
Stumbled on this [StackOverflow answer](https://stackoverflow.com/a/34800667/13138783)
https://github.com/synfig/synfig/pull/1525#issuecomment-653833203

Better and intelligent CMake code to copy:
```cmake
configure_file(logo.sif logo.sif COPYONLY)
```

**input** is automatically mapped to `CMAKE_CURRENT_SOURCE_DIR`
and **output** to `CMAKE_CURRENT_BINARY_DIR`
so no need to explicitly enter the path variables :)

AYESDIE replied:
`configure_file` is used to work with `*.in` files to substitute variable values referenced in the file content. Yes, copying is an alternate use of `configure_file`, however, `file(COPY ...)` preserves timestamps of input file as well. 

[CMake configure_file](https://cmake.org/cmake/help/v3.0/command/configure_file.html)
> Copies a file <input> to file <output> and substitutes variable values referenced in the file content.

[CMake file](https://cmake.org/cmake/help/v3.0/command/file.html#copy)
> Copying preserves input file timestamps, and optimizes out a file if it exists at the destination with the same timestamp.

You can check one use of `configure_file` [here](https://github.com/synfig/synfig/blob/master/synfig-studio/images/CMakeLists.txt#L276).

Keyikedalube replied:
https://github.com/synfig/synfig/pull/1525#issuecomment-653851114

AYESDIE commented:
Changes:
- Added a basic CPackConfig.cmake script and PACKAGING.md to generate .deb packages.

- Fixes #1402.

- Updated `SYNFIG_INTLTOOL_MERGE` to account for .desktop.in files.

- Made changes to move icon to the specified location in cmake builds.

- Added dependencies on `synfig_bin` and `synfigstudio` to make building process easier.

So now, building can be done by:
```
mkdir build && cd build
cmake ../
make package -j8
```
This will generate a *.deb* package in the build folder.
So now, no need to build targets like `build_images` or `appdata_xml` on our own.

ice0 commented on synfig-studio/CMakeLists.txt:
```diff
@@ -15,3 +16,35 @@ add_subdirectory(plugins)
+    INSTALL_DESTINATION "share/applications"
+)
+
+add_custom_command(
```
Better to do this part, like Inkscape did:
https://gitlab.com/inkscape/inkscape/-/blob/master/org.inkscape.Inkscape.desktop.template

AYESDIE replied:
They used `configure_file` in a way we used it to configure `splash_screen_development.sif.in` where it replaces the `${INKSCAPE_ICONPATH}` with the value of that variable. Making changes to this would also require changes in autotools script, and I'm not too used with scripting in that.

ice0 replied:
Agree.

ice0 commented on cmake/SynfigIntltool.cmake:
```diff
@@ -1,26 +1,38 @@
 find_program(INTLTOOL_MERGE intltool-merge)
 
 function(SYNFIG_INTLTOOL_MERGE)
```
Perhaps it is better to rename it to something with `STUDIO` in the name, so that it indicates that this function can only be used for a studio project?

ice0 commented:
Merged. Thank you!
